packages/service-library/src/servicelib/fastapi/tracing.py

```diff
@@ -127,3 +136,66 @@ def setup_tracing(
 
 def setup_httpx_client_tracing(client: AsyncClient | Client):
     HTTPXClientInstrumentor.instrument_client(client)
+
+
+def _opentelemetry_function_span(func: Callable):
+    """Decorator that wraps a function call in an OpenTelemetry span."""
+    tracer = trace.get_tracer(__name__)
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        with tracer.start_as_current_span(f"{func.__module__}.{func.__name__}"):
+            return func(*args, **kwargs)
+
+    @wraps(func)
+    async def async_wrapper(*args, **kwargs):
+        with tracer.start_as_current_span(f"{func.__module__}.{func.__name__}"):
+            return await func(*args, **kwargs)
+
+    if inspect.iscoroutinefunction(func):
+        return async_wrapper
+    else:
+        return wrapper
+
+
+def _opentelemetry_method_span(cls):
```

Comment:
here I have some questions:
- how heavy is using this? If I get it right, using this wraps every single function with a network call to the telemetry collector right? That means any call is network-limited is that correct?
- if yes, then what happens if the collector is broken or the network is shaky?
- does this break the function?
- if there are network issues, does it retry? how many times? what is the timeout?
- if you telemetrize a ping function for example, that might be quite bad no? these function typically should run with no "brakes", can you exclude functions?